contrib/pg_stat_statements/output/test_pg_stat_statements.source

```diff
@@ -0,0 +1,139 @@
+SELECT pg_stat_statements_reset();
+ pg_stat_statements_reset 
+--------------------------
+ 
+(1 row)
+
+SELECT GROUPING (a) FROM t GROUP BY ROLLUP(a, b);
+ grouping 
+----------
+(0 rows)
+
+-- launch not equivalent query
+SELECT GROUPING (b) FROM t GROUP BY ROLLUP(a, b);
+ grouping 
+----------
+(0 rows)
+
+-- check that 2 queries have separate entries
+SELECT query, calls FROM pg_stat_statements ORDER BY query;
+                       query                       | calls 
+---------------------------------------------------+-------
+ SELECT GROUPING (a) FROM t GROUP BY ROLLUP(a, b); |     1
+ SELECT GROUPING (b) FROM t GROUP BY ROLLUP(a, b); |     1
+ SELECT pg_stat_statements_reset();                |     1
+(3 rows)
+
+-- check that different grouping options result in separate entries
+SELECT COUNT (*) FROM t GROUP BY ROLLUP(a, b);
+ count 
+-------
+(0 rows)
+
+SELECT COUNT (*) FROM t GROUP BY CUBE(a, b);
+ count 
+-------
+(0 rows)
+
+SELECT COUNT (*) FROM t GROUP BY GROUPING SETS(a, b);
+ count 
+-------
+(0 rows)
+
+SELECT COUNT (*) FROM t GROUP BY GROUPING SETS((a), (a, b));
+ count 
+-------
+(0 rows)
+
+SELECT COUNT (*) FROM t GROUP BY a, b;
+ count 
+-------
+(0 rows)
+
+SELECT query, calls FROM pg_stat_statements ORDER BY query;
+                            query                             | calls 
+--------------------------------------------------------------+-------
+ SELECT COUNT (*) FROM t GROUP BY a, b;                       |     1
+ SELECT COUNT (*) FROM t GROUP BY CUBE(a, b);                 |     1
+ SELECT COUNT (*) FROM t GROUP BY GROUPING SETS((a), (a, b)); |     1
+ SELECT COUNT (*) FROM t GROUP BY GROUPING SETS(a, b);        |     1
+ SELECT COUNT (*) FROM t GROUP BY ROLLUP(a, b);               |     1
+ SELECT GROUPING (a) FROM t GROUP BY ROLLUP(a, b);            |     1
+ SELECT GROUPING (b) FROM t GROUP BY ROLLUP(a, b);            |     1
+ SELECT pg_stat_statements_reset();                           |     1
+ SELECT query, calls FROM pg_stat_statements ORDER BY query;  |     1
+(9 rows)
+
+-- check several parameters options in ROLLUP
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+ pg_stat_statements_reset 
+--------------------------
+ 
+(1 row)
+
+SELECT COUNT (*) FROM t GROUP BY ROLLUP(a, b);
+ count 
+-------
+(0 rows)
+
+SELECT COUNT (*) FROM t GROUP BY ROLLUP(b);
+ count 
+-------
+(0 rows)
+
+SELECT query, calls FROM pg_stat_statements ORDER BY query;
+                     query                      | calls 
+------------------------------------------------+-------
+ SELECT COUNT (*) FROM t GROUP BY ROLLUP(a, b); |     1
+ SELECT COUNT (*) FROM t GROUP BY ROLLUP(b);    |     1
+ SELECT pg_stat_statements_reset();             |     1
+(3 rows)
+
+-- check group_id() in query
+SELECT group_id() AS c FROM t GROUP BY GROUPING SETS ((a), (a,b));
+ c 
+---
+(0 rows)
+
+-- check anytable parameter for a function
+CREATE FUNCTION multiset_1(a anytable) RETURNS TABLE(a int, b text)
```

Comment:
Try using anytable_out as a multiset function. Its call always fails, so you need to make a query so that it is not called but is in the query.
Something like that. Is it possible to reproduce the problem with a similar query?
```sql
select count(1) from anytable_out( table(select 1) ) where 1 = 0;
```

Reply:
Indeed, it can help to simplify the test. Updated it.